### PR TITLE
reference/security: add certificate-based authentication doc

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

Add the certificate-based authentication document.

**This feature will apply to `v3.1` after `v3.1.0-beta.1` is released.**

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/2153

### Which version does your change affect? 

dev, v3.0

### CODE REVIEW

File: dev/reference/security/cert-based-authentication.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+
+```bash
+mysql -utest -h0.0.0.0 -P4000 --ssl-cert /path/to/client-cert.new.pem --ssl-key /path/to/client-key.new.pem --ssl-ca /path/to/ca-cert.pem
+```
+
+## Configure the user certificate information for login verification
+
+Connect TiDB using the client to configure the login authentication. You can configure the user certificate information for login verification in the following three methods:
+
++ Configure the certificate when creating a user (`create user`):
```
Comment (yikeke) [inline dev/reference/security/cert-based-authentication.md]:
```suggestion
+ Configure the certificate information to be verified when creating a user (`create user`):
```

File: dev/reference/security/cert-based-authentication.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    create user 'u1'@'%'  require issuer '/C=US/ST=California/L=San Francisco/O=PingCAP Inc./OU=TiDB/CN=TiDB admin/emailAddress=s@pingcap.com' subject '/C=US/ST=California/L=San Francisco/O=PingCAP Inc./OU=TiDB/CN=tpch-user1/emailAddress=zz@pingcap.com' cipher 'TLS_AES_256_GCM_SHA384';
+    grant all on *.* to 'u1'@'%';
+    ```
+
++ Configure the certificate when granting privilege:
```
Comment (yikeke) [inline dev/reference/security/cert-based-authentication.md]:
```suggestion
+ Configure the certificate information to be verified when granting privileges:
```
s/privilege/privileges

File: dev/reference/security/cert-based-authentication.md [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    create user 'u1'@'%';
+    grant all on *.* to 'u1'@'%' require issuer '/C=US/ST=California/L=San Francisco/O=PingCAP Inc./OU=TiDB/CN=TiDB admin/emailAddress=s@pingcap.com' subject '/C=US/ST=California/L=San Francisco/O=PingCAP Inc./OU=TiDB/CN=tpch-user1/emailAddress=zz@pingcap.com' cipher 'TLS_AES_256_GCM_SHA384';
+    ```
+
++ Configure the certificate when altering a user:
```
Comment (yikeke) [inline dev/reference/security/cert-based-authentication.md]:
```suggestion
+ Configure the certificate information to be verified when altering a user:
```

File: dev/reference/security/cert-based-authentication.md [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    alter user 'u1'@'%' require issuer '/C=US/ST=California/L=San Francisco/O=PingCAP Inc./OU=TiDB/CN=TiDB admin/emailAddress=s@pingcap.com' subject '/C=US/ST=California/L=San Francisco/O=PingCAP Inc./OU=TiDB/CN=tpch-user1/emailAddress=zz@pingcap.com' cipher 'TLS_AES_256_GCM_SHA384';
+    ```
+
+In the above three methods, `require subject`, `require issuer`, and `require cipher` are respectively used to check the X509 certificate attributes. You can configure one item or multiple items using the space or `and` as the separator.
```
Comment (yikeke) [inline dev/reference/security/cert-based-authentication.md]:
```suggestion
In the above three methods, `require subject`, `require issuer`, and `require cipher` are used to check the X509 certificate attributes. You can configure one item or multiple items using the space or `and` as the separator.
```

File: dev/reference/security/cert-based-authentication.md [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+
+4. Generate the combined CA certificate:
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    cat ca-cert.new.pem ca-cert.old.pem > ca-cert.pem
+    ```
+
+After this update, use the combined CA certificate and restart the TiDB server. Then the server accepts both the new and old CA certificates.
```
Comment (yikeke) [inline dev/reference/security/cert-based-authentication.md]:
```suggestion
After the above operations, restart the TiDB server with the newly created combined CA certificate. Then the server accepts both the new and old CA certificates.
```

File: dev/reference/security/cert-based-authentication.md [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    mysql -utest -h0.0.0.0 -P4000 --ssl-cert /path/to/client-cert.new.pem --ssl-key /path/to/client-key.new.pem --ssl-ca /path/to/ca-cert.pem
+    ```
+
+### Update the server key and certificate
+
+1. Generate the new RSA key of the sever:
```
Comment (yikeke) [inline dev/reference/security/cert-based-authentication.md]:
```suggestion
1. Generate the new RSA key of the server:
```

File: dev/reference/security/cert-based-authentication.md [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+{{< copyable "" >}}
+
+```
+[security]
+ssl-cert ="path/to/server-cert.pem"
+ssl-key ="path/to/server-key.pem"
+ssl-ca="path/to/ca-cert.pem"
+```
+
+Start TiDB logs. If the following information is displayed in the log, the configuration is successful:
```
Comment (lysu) [inline dev/reference/security/cert-based-authentication.md]: maybe better some thing like "start TiDB and watch logs"

**Pull request merged**